extensions/azurePublishNew/src/node/getResources.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IBotProject } from '@botframework-composer/types';
+
+import {
+  getProvisionServices,
+  GetResourcesResult,
+  ResourceDefinition,
+  ResourceProvisionService,
+} from './provisionService';
+import { appRegistrationDefinition } from './azureResources/appRegistration';
+import { webAppResourceDefinition } from './azureResources/webApp';
+import { azureFunctionsDefinition } from './azureResources/azureFunctions';
+import { botRegistrationDefinition } from './azureResources/botChannel';
+import { luisAuthoringDefinition } from './azureResources/luisAuthoring';
+import { appInsightsDefinition } from './azureResources/appInsights';
+import { cosmosDbDefinition } from './azureResources/cosmosDb';
+import { luisPredictionDefinition } from './azureResources/luisPrediction';
+import { blobStorageDefinition } from './azureResources/blobStorage';
+import { qnaDefinition } from './azureResources/qna';
+import { servicePlanDefinition } from './azureResources/servicePlan';
+
+export const AZURE_HOSTING_GROUP_NAME = 'Azure Hosting';
+export const COGNITIVE_SERVICES_GROUP_NAME = 'Cognitive Services';
+// export type HostingGroupName = 'Azure Hosting' | 'Cognitive Services';
+
+const availableResources: ResourceDefinition[] = [
+  appRegistrationDefinition,
+  webAppResourceDefinition,
+  botRegistrationDefinition,
+  azureFunctionsDefinition,
+  cosmosDbDefinition,
+  appInsightsDefinition,
+  luisAuthoringDefinition,
+  luisPredictionDefinition,
+  blobStorageDefinition,
+  qnaDefinition,
+  servicePlanDefinition,
+];
+
+export const getResources = (project: IBotProject): GetResourcesResult[] => {
+  const provisionServices: Record<string, ResourceProvisionService> = getProvisionServices();
```

Comment:
The plug-in interface is an instance where the getresources and provision methods have a shared lifeltime. Consider that we don't want to create the provision services here and in provision.  I might pass the list of services as a parameter.  You might end up with a provisioning.ts that has the getResources and provision methods and a constants.ts that contains the list of definitions and services.  Keeping those constants together can be good because they should be updated in lock step.